File: ruby/test/sorts/bubble_sort_test.rb
require_relative '../../src/sorts/bubble_sort'
require_relative '../../src/util/util'
require_relative '../../src/util/constants'
require_relative '../util'

require 'test_helper'
require 'minitest/autorun'
require 'minitest/spec'

describe BubbleSort do
    describe 'when passed an array' do
        it 'should correctly sort it' do
            arr = TestUtil.random_array(Constants.ARRAY_SIZE)

            result = BubbleSort.sort(arr)

            Util.is_sorted(result).must_equal true
        end
    end
end

File: python/structures/test/sorts/bubble_sort_test.py
import unittest

import structures.src.sorts.bubble as bubble_sort
import structures.src.util.sort_util as util
import structures.src.util.constants as const

class TestBubbleSort(unittest.TestCase):
    def test_sorted(self):
        arr = util.random_array(const.ARRAY_SIZE)

        # make sure the starting array isn't sorted
        self.assertFalse(util.is_sorted(arr))

        # sort the array
        result = bubble_sort.sort(arr)

        # make sure it worked
        self.assertTrue(util.is_sorted(result))

if __name__ == '__main__':
    unittest.main()
